pig/src/main/java/org/jboss/pnc/bacon/pig/impl/repo/RepoManager.java

```diff
@@ -520,65 +531,46 @@ private void resolveAndRepackage(
                                     a.getClassifier(),
                                     a.getVersion()))
                     .forEach(extensionRtArtifactList::add);
-        } else {
-            resolveSet = GavSet.includeAll();
         }
 
         final List<Exclusion> transitiveExclusions = parseExclusions(params.get("excludeTransitive"));
 
         ensureDefaultJarsIncluded(extensionRtArtifactList);
 
+        final List<Artifact> artifactList = ensureVersionsSet(extensionRtArtifactList);
         if (log.isDebugEnabled()) {
             log.debug(
-                    "About to resolve artifacts \n" + extensionRtArtifactList.stream()
-                            .map(Artifact::toString)
-                            .collect(Collectors.joining("\n")));
+                    "About to resolve artifacts \n"
+                            + artifactList.stream().map(Artifact::toString).collect(Collectors.joining("\n")));
             if (!transitiveExclusions.isEmpty()) {
                 log.debug("Applying exclusions to transitive dependencies:");
                 transitiveExclusions.forEach(e -> log.debug("  " + e));
             }
         }
 
-        Map<Artifact, String> redhatVersionExtensionArtifactMap = collectRedhatVersions(extensionRtArtifactList);
-
         final List<Dependency> managedDeps = new ArrayList<>(bomConstraints);
         final StringBuilder bannedReport = new StringBuilder();
         final Set<ArtifactCoords> collectedArtifacts = new HashSet<>();
-        for (Artifact extensionRtArtifact : extensionRtArtifactList) {
+        for (Artifact artifact : artifactList) {
             // this will resolve all the artifacts and their dependencies and as a consequence populate the local
             // Maven repo
             // specified in the user settings.xml
-            String aptVersion = redhatVersionExtensionArtifactMap
-                    .getOrDefault(extensionRtArtifact, extensionRtArtifact.getVersion());
-            if (extensionRtArtifact.getArtifactId().contains("plugin")) {
-                Artifact pluginArtifact = new DefaultArtifact(
-                        extensionRtArtifact.getGroupId(),
-                        extensionRtArtifact.getArtifactId(),
-                        extensionRtArtifact.getExtension(),
-                        aptVersion);
-                log.debug("Resolving dependencies of plugin {}", pluginArtifact);
-                mvnResolver.resolvePluginDependencies(pluginArtifact);
+            if (artifact.getArtifactId().contains("maven-plugin")) {
```

Comment:
I don't think this is matching same artifacts as before. Some plugin artifacts doesn't have "maven-plugin" in name, is it intentional?

Reply:
Yes, it is intentional. I found that some artifacts had `plugin` in the artifactId name that weren't a Maven plugin and their dependencies weren't properly resolved.

Reply:
Either way, this check is a hack, unfortunately, and needs to be handled in some better way.
A proper way would be to have a dedicated config option for artifacts that are Maven plugins or this code has to resolve the artifact and check whether it contains Maven plugin metadata, which will be less efficient but more correct.

Reply:
As the rest of it, this is a Quarkus-specific part. The reason Maven plugin dependencies are resolved differently from the rest of the artifacts is that, the BOM configured as the source of dependencies to resolve should not be enforced on Maven plugin artifacts, as it wouldn't happen in customer applications. But for the rest of the artifacts the BOM has to be enforced while resolving dependencies.